helix-core/src/line_ending.rs

```diff
@@ -151,6 +170,12 @@ pub fn get_line_ending_of_str(line: &str) -> Option<LineEnding> {
     }
 }
 
+pub fn set_line_ending_of_str(doc: &str, line_ending: LineEnding) -> String {
+    doc.lines()
+        .map(|line| format!("{}{}", line, line_ending.as_str()))
+        .collect()
+}
```

Comment:
Extremely inefficient, please use https://doc.rust-lang.org/std/primitive.str.html#method.replace

Reply:
(You can drop this whole method and inline the call to `replace`)

Reply:
Ah right, it might use mixed line endings, hmm...